packages/financial-templates-lib/src/price-feed/UniswapPriceFeed.js

```diff
@@ -117,6 +121,7 @@ class UniswapPriceFeed extends PriceFeedInterface {
     this.currentTwap = this._computeTwap(this.events, currentTime - this.twapLength, currentTime);
 
     this.lastUpdateTime = currentTime;
+    this.earliestTime = this.events[0].timestamp;
```

Comment:
Why not `currentTime - this.historicalLookback`? Also, should we compute this on-demand in the getEarliestTime function rather than using a state variable?

I only ask because this would separate an _expected_ lack of data due to insufficient time range from the case where data is missing because of some event/querying issue.

Reply:
We can also use `currentTime - this.historicalLookBack` but I thought getting the last block in the block history's timestamp would be an improvement.

> Also, should we compute this on-demand in the getEarliestTime function rather than using a state variable?

This is a fair point, I'll make that change.

Reply:
Actually thinking about this more, I feel like `currentTime - this.historicalLookback` is more reasonable to the user and hides implementation details. It can also be consistent across price feeds.